src/include/distributed/extended_op_node_utils.h

```diff
@@ -0,0 +1,39 @@
+/*-------------------------------------------------------------------------
+ *
+ * extended_op_node_utils.h
+ *	  General Citus planner code.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef EXTENDED_OP_NODE_UTILS_H_
+#define EXTENDED_OP_NODE_UTILS_H_
+
+#include "distributed/multi_logical_planner.h"
+
+
+/*
+ * SharedSourceForExtendedOpNodes is a helper structure that is used to
+ * share the common information among the worker and master extended
+ * op nodes.
+ *
+ * It is designed to be a read-only singleton object per extended op node
+ * generation and processing.
+ */
+typedef struct SharedSourceForExtendedOpNodes
+{
+	bool groupedByDisjointPartitionColumn;
+	bool repartitionSubquery;
+	bool hasNonPartitionColumnDistinctAgg;
+	bool pullDistinctColumns;
+	bool pushDownWindowFunctions;
+} SharedSourceForExtendedOpNodes;
+
+
+extern SharedSourceForExtendedOpNodes BuildSharedSourceForExtendedOpNodes(
```

Comment:
`BuildSharedSourceForExtendedOpNodes ` and `SharedSourceForExtendedOpNodes `  are probably not the best names. I'm still trying to come-up with a better alternative. Any suggestion?

Reply:
This was the first thing to itch my eyes. 
something shorter like `ExtendedOpNodeStats` or `ExtendedOpNodeInfo` or even `ExtendedOpNodeMetadata` might be more meaningful than `SharedSource...`

Reply:
I intentionally made the name so long that we have to change it before merging in 😄 

 I actually considered `ExtendedOpNodeStats `. But, it might not be compatible with future changes in case we add some more stuff here? What do you think?

Also, what about `ExtendedOpNodeSummary`?

Reply:
> ExtendedOpNodeStats

@mtuncer I'm planning to continue with `ExtendedOpNodeStats ` which makes more sense now.